src/pypowsybl_jupyter/selectcontext.py

```diff
@@ -32,6 +34,7 @@ def get_vls(self):
     def set_selected(self, id):
         if id in self.vls.index:
             self.selected_vl = id
+            self.add_to_history(id)
```

Comment:
Then only if not coming from history.
Maybe there should be two methods, something like one `set_selected_history` and one `set_selected_found`? and similarly for `update_explorer`

Reply:
logic changed